source/microbit/modneopixel.cpp

```diff
@@ -34,45 +34,57 @@ extern "C" {
 #include "py/runtime.h"
 #include "microbit/modmicrobit.h"
 
-// This is the pixel colour ordering
-#define COL_NUM_COMPONENTS (3)
+// This is the pixel channel colour ordering       |R G B W|
+// WS8212/WS8212B chips use GRB-W ordering, hence: |1 0 2 3|
 #define COL_IDX_RED (1)
 #define COL_IDX_GREEN (0)
 #define COL_IDX_BLUE (2)
+#define COL_IDX_WHITE (3)
 
+// sends bytes to pixels
 // This is implemented in assembler to get the right timing
-extern void sendNeopixelBuffer(uint32_t pin, uint8_t* data_address, uint16_t num_leds);
+extern void sendNeopixelBuffer(uint32_t pin, uint8_t* data_address, uint16_t num_leds, uint16_t channels);
 
 extern const mp_obj_type_t neopixel_type;
 
+//Input args when calling neopixel.NeoPixel()?
 typedef struct _neopixel_obj_t {
     mp_obj_base_t base;
     uint16_t pin_num;
     uint16_t num_pixels;
+    uint16_t num_channels;
     uint8_t data[];
 } neopixel_obj_t;
 
 STATIC mp_obj_t neopixel_show_(mp_obj_t self_in);
 
 static inline void neopixel_clear_data(neopixel_obj_t *self) {
-    memset(&self->data[0], 0, COL_NUM_COMPONENTS * self->num_pixels);
+    memset(&self->data[0], 0, self->num_channels * self->num_pixels);
 }
 
 STATIC mp_obj_t neopixel_make_new(const mp_obj_type_t *type_in, mp_uint_t n_args, mp_uint_t n_kw, const mp_obj_t *args) {
     (void)type_in;
-    mp_arg_check_num(n_args, n_kw, 2, 2, false);
+    mp_arg_check_num(n_args, n_kw, 3, 3, false);
```

Comment:
this change would break existing use of this class, because it now requires the `num_channels` arg